### ISSUE TITLE
Server gets hit twice when filtering on a column

### ISSUE THREAD

**Issue by anujism** [wenzhixin/bootstrap-table#2417]
Is it possible to add a delay (table or column level) so that it will send request to server only after a few seconds (and only with the text present at that time).

**Comment by anujism** [comment]
I found it in another issue. That is data-search-time-out. But now I am facing a problem that whenever I search in a column, it hits the server twice. When I debugged the issue I found this:

BootstrapTable.prototype.onColumnSearch = function (event) {
        copyValues(this);
        var text = $.trim($(event.currentTarget).val());
        var $field = $(event.currentTarget).closest('[data-field]').data('field');

        if ($.isEmptyObject(this.filterColumnsPartial)) {
            this.filterColumnsPartial = {};
        }
        if (text) {
            this.filterColumnsPartial[$field] = text;
        } else {
            delete this.filterColumnsPartial[$field];
        }

        this.options.pageNumber = 1;
        this.onSearch(event);
        this.updatePagination();
        this.trigger('column-search', $field, text);
    };

Here, The server is called at this.onSearch(event) statement and then again on this.updatePagination().

I have data-side-pagination set as "server".

**Comment by wenzhixin** [comment]
Please provide a [jsFiddle](https://github.com/wenzhixin/bootstrap-table/blob/develop/CONTRIBUTING.md#bug-reports) to show your problem, thanks!

**Comment by anujism** [comment]
Please see this: http://jsfiddle.net/e3nk137y/7118/
When are you planning to release the next version of filter-control?

**Comment by wenzhixin** [comment]
@djhvscf

**Comment by djhvscf** [comment]
Let me check it

**Comment by djhvscf** [comment]
This issue is already fixed in this branch https://github.com/djhvscf/bootstrap-table/blob/develop/src/extensions/filter-control/bootstrap-table-filter-control.js
